### PR TITLE
Fix a bug in the crawler OGlobo (issue #80) 

### PR DESCRIPTION
**Pull request by lucasmachadorj**
If published time occasionally may be in a different format (%Y-%m-%d), we can use a try/except block to extract in both formats.

### CODE REVIEW

**Comment by fccoelho** [comment]
It seems to be ok for this crawler, in downloader.py I first run it through dateutil's parsedate function, which can solve most common variations. Then, I proceed to my own guesses, but I always have a fallback solution, if all else fails, like using the date of the capture, if I am capturing the latest news.

**Comment by flavioamieiro** [comment]
I also like @fccoelho 's idea of using [dateutil.parser.parse](https://dateutil.readthedocs.org/en/latest/parser.html#dateutil.parser.parse) with a default. I think it will be more robust and also hide this complexity inside `dateutil`.

`published_time = dateutil.parser.parse(published_time_str, default=datetime.date.today())` show be enough to replace this code and, if the date is not there, use today's date.

**Comment by lucasmachadorj** [comment]
I've used dateutil as suggested in globo's crawler and changed estadao's crawler to save article in database only if content's body is not blank.

**Comment by lucasmachadorj** [comment]
Some news pages in Folha de São Paulo are just an introduction to a blog's topic that is contained in another page. I've chosen to disregard this articles and go on. Another approach is to give a BeautifulSoup object as input to extract_content() method and looking for a link whose text is "blog". If it exists, we could get the body content from this link. Any suggestion?

**Comment by fccoelho** [comment]
I think that it's worth trying to get the full content. But we may have to store the new link. Perhaps replacing the old one?

**Comment by flavioamieiro** [comment]
I agree with @fccoelho. It's worth getting the full content. And I think looking for a link whose text is "blog" (after you have already detected that the page is one of those) might be a good idea.

**Comment by lucasmachadorj** [comment]
@flavioamieiro I passed 2 to find_articles due some tests and have forgotten to push fix. Now it saves the articles in database and deals with http or https protocol

**Comment by flavioamieiro** [comment]
When I ran `crawler_oglobo.py` I got the following traceback:

    $ python crawler_oglobo.py 
    Traceback (most recent call last):
      File "crawler_oglobo.py", line 100, in <module>
        ARTICLES.insert(article, w=1)
      File "/home/amieiro/.virtualenvs/mediacloud_backend/local/lib/python2.7/site-packages/pymongo/collection.py", line 415, in insert
        self.uuid_subtype, client)
    bson.errors.InvalidDocument: Cannot encode object: datetime.date(2015, 7, 30)

This is because [pymongo does not support saving date objects](http://api.mongodb.org/python/current/faq.html#how-can-i-save-a-datetime-date-instance). You need to use a datetime object in the document you want to save.

**Comment by flavioamieiro** [comment]
Also, in [`crawler_oglobo.py`](https://github.com/lucasmachadorj/mediacloud_backend/blob/master/capture/crawler_oglobo.py#L73-L76) the exception classes `ConnectionError` and `Timeout` are not defined. This didn't trigger a `NameError` when I ran the code because I didn't get one of those exceptions.

File: capture/crawler_folha.py [hunk 1]
```diff
@@ -1,9 +1,140 @@
...
+        logger.error("Failed to fetch:{0}".format(url))
+        return None
+    except Timeout:
+        logger.error("Timed out while fetching {0}".format(url))
+        return None
+
+    extractor = Goose({'use_meta_language': False, 'target_language':'pt'})
+    news = extractor.extract(url=url)
+
+    article['link_content'] = compress_content(response_content)
```
Comment (flavioamieiro) [inline capture/crawler_folha.py]: `response_content` doesn't exist here. Should it be `news`?

Comment (lucasmachadorj) [inline capture/crawler_folha.py]: It should be response.content, but is better to use response.text as you have already suggested.

**Comment by lucasmachadorj** [comment]
@flavioamieiro About oglobo's crawler, I've fixed the problem in get_published_time method, then I noticed that in some news the published time is still none, even with the default datetime. I don't know why it still returns none, but I noticed that all webpages with this problem are not articles but slideshows or similar, then should not be saved in database, not least because goose extractor takes random content. So when saving in database, I've disregarded article with content or published time none.

**Comment by lucasmachadorj** [comment]
@flavioamieiro Please, look at the crawler_folha and analyze the changes I've made in extract_content for getting the full content. @fccoelho  I decided to replace old url by new one.

**Comment by flavioamieiro** [comment]
@lucasmachadorj I've read and ran the code again, and everything looks OK. I think we should merge. @fccoelho any objections?

**Comment by fccoelho** [comment]
No objections, please go ahead, merge and put in production.

**Pull request merged**